src/adapter/src/coord/sequencer/cluster.rs

```diff
@@ -450,9 +457,32 @@ impl Coordinator {
             },
         };
 
-        let id = self.catalog_mut().allocate_user_replica_id().await?;
+        let cluster = self.catalog().get_cluster(cluster_id);
+
+        // Only internal users can add replicas to managed clusters
+        if cluster.is_managed() && !session.user().is_internal() {
+            // TODO: PlanError isn't great to use here.
+            return Err(PlanError::ManagedCluster {
+                cluster_name: cluster.name().to_string(),
+            }
+            .into());
+        }
+        // Internal users have to specify BILLED AS when creating replicas, but nobody
+        // else can specify BILLED AS
+        if let ReplicaLocation::Managed(ManagedReplicaLocation { billed_as, .. }) = &config.location
+        {
+            match (session.user().is_internal(), billed_as) {
+                (true, None) if cluster.is_managed() => coord_bail!(
```

Comment:
Why must it be specified by system users? So that employees don't accidentally create billed replicas? If that's the case, does it make sense to only allow free? When would an employee need to create a billed non-free replica?